### PR TITLE
add `opts.compiler` allows third party to override the internal compiler

### PR DESCRIPTION
**Pull request by mattma**
I read the comments that @lazd made in other posts. Here is my version of rolling out the User custom compiler.

### CODE REVIEW

**Comment by alexgraul** [comment]
Sorry to nag but any chance of getting this merged soon?

**Comment by BowlingX** [comment]
+1 for merging this very soon :)

**Comment by lazd** [comment]
Sorry for the delay here folks. I need to do a little work in terms of documentation, examples, etc after merging this and I haven't yet found the time to do that yet. I'll get this in and make sure we have a perfect solution for compiling templates for Ember with HTMLBars soon.

In the mean time, you can point your `package.json` at @mattma's fork:

```json
"dependencies": {
  "gulp-handlebars": "git+ssh://git@github.com:mattma/gulp-handlebars.git#7789b4475dfc981f705fd2917ec05da117a0d35c"
}
```

File: README.md [hunk 1]
```diff
@@ -261,6 +261,11 @@ Type: `Object`
 
 Handlebars library to use for precompilation. By default, the latest stable version of Handlebars is used.
 
+#### options.compiler
```
Comment (lazd) [inline README.md]: Let's go ahead and do this after merge.

File: test/main.js [hunk 2]
```diff
@@ -128,5 +128,34 @@ describe('gulp-handlebars', function() {
...
+        should.exist(newFile.contents);
+
+        // @TODO: it should call `fileMatchesExpected` function
+        // But I got an error of  expected: "BASIC TEMPLATE\n", actual: "BASIC TEMPLATE"
+        // so that I manually adding the newLine character.
+        // I think it related to plain text in the 'Basic_htmlbars.js'
+        var expected = getExpectedString('Basic_htmlbars.js');
+        var tested = String(newFile.contents)+'\n';
+        tested.should.equal(expected);
+        // fileMatchesExpected(newFile, 'Basic_htmlbars.js', 'Basic.js');
```
Comment (lazd) [inline test/main.js]: Let's correct this after merge.

**Pull request merged**